packages/itwinui-react/src/core/Panels/Panels.tsx

```diff
@@ -0,0 +1,600 @@
+   * - `id: string` - The panel to show.
+   * - `direction?: 'prev' | 'next' | 'undefined'`:
+   *   - animates to the new panel in the specified direction or transitions instantly if `undefined`.
+   *   - moves focus to the new panel when `next` or `undefined` and to the trigger when `prev`.
+   */
+  activeId?: CurrentPanelId;
+  /** Useful for controlled mode */
+  onActiveIdChange?: (newActiveId: CurrentPanelId) => void;
+  children: React.ReactNode;
+  /**
+   * Pass an instance created by `useInstance` to control the panels imperatively.
+   *
+   * @example
+   * const panels = Panels.useInstance();
+   * <Panels instance={panels} />
+   */
+  instance?: PanelsInstance;
+  /**
+   * To customize the panel transition animations, pass one of the following:
+   * - `number`: Duration in milliseconds. Overrides the default `animationOptions`.
+   * - `object`: For more customization. Merges with the default `animationOptions`.
+   * - `null`: Disables animations completely.
+   *
+   * @see `HTMLElement.animate`
+   */
+  animationOptions?: Parameters<HTMLElement['animate']>[1] | null;
+};
+
+/**
+ * Component that manages the logic for layered panels.
+ * It can be used anywhere to create layers. E.g. `Menu`, `InformationPanel`, `Popover`, etc.
+ *
+ * Notes:
+ * - There can only be one panel with triggers to a particular other panel. i.e. Panel Y should have triggers only in
+ * one particular panel X (only one clear back navigation). Panel X can have multiple triggers that point to panel Y.
+ *
+ * @example
+ * <Panels.Wrapper
+ *   initialActiveId={panelIdRoot}
+ *   as={Surface}
+ *   style={{ inlineSize: '300px', blockSize: '500px' }}
+ * >
+ *   <Panels.Panel id={panelIdRoot} as={List}>
+ *     <Surface.Header as={Panels.Header}>Base</Surface.Header>
+ *     <ListItem>
+ *       <Panels.Trigger for={panelIdMoreInfo}>
+ *         <ListItem.Action>More details</ListItem.Action>
+ *       </Panels.Trigger>
+ *     </ListItem>
+ *   </Panels.Panel>
+ *
+ *   <Panels.Panel
+ *     id={panelIdMoreInfo}
+ *     as={Flex}
+ *     flexDirection='column'
+ *     alignItems='stretch'
+ *   >
+ *     <Surface.Header as={Panels.Header}>More details</Surface.Header>
+ *     …content
+ *   </Panels.Panel>
+ * </Panels.Wrapper>
+ */
+const PanelsWrapper = React.forwardRef((props, forwardedRef) => {
+  const {
+    initialActiveId,
+    children,
+    activeId: activeIdProp,
+    className,
+    onActiveIdChange,
+    instance,
+    animationOptions: animationOptionsProp,
+    ...rest
+  } = props;
+
+  const panelElements = React.useRef<Record<string, HTMLElement | null>>({});
+
+  const [currentPanelId, setCurrentPanelId] = React.useState<CurrentPanelId>({
+    id: initialActiveId,
+  });
+
+  // Reducer to manage panel animations
+  const [{ animations }, dispatch] = React.useReducer(panelAnimationReducer, {
+    currentPanelId,
+    setCurrentPanelId,
+  } satisfies PanelAnimationState);
+
+  const animateToPanel = React.useCallback(
```

Comment:
There is so much animation-related code in this component that it's making the main component code difficult to understand.

Is it possible to write the component in a way that the general flow of the component is similar, regardless of whether we are animating or not? e.g. important component code should not be so tightly coupled with the animation code.